Src/IronPython/Compiler/Ast/ClassDefinition.cs

```diff
@@ -138,7 +138,10 @@ public class ClassDefinition : ScopeStatement {
             }
 
             // Try to bind in outer scopes, except the global scope
-            for (ScopeStatement parent = Parent; parent is not null && !parent.IsGlobal; parent = parent.Parent) {
+            for (ScopeStatement? parent = Parent;
```

Comment:
I spent entirely too much time staring at this to wrap my head around the logic. I think in this case it could have been reduced to `for (ScopeStatement parent = Parent; !parent.IsGlobal; parent = parent.Parent)`? But I guess it's to be consistent with the other outer binding loops? Wondering if it's worth having a helper (I think this works?):
```C#
internal bool TryBindOuterScopes(ScopeStatement from, PythonReference reference, [NotNullWhen(true)] out PythonVariable? variable, bool stopAtGlobal = false) {
    for (ScopeStatement? parent = from.Parent; parent != null && !(stopAtGlobal && parent.IsGlobal); parent = !parent.IsGlobal ? parent.Parent : null) {
        if (parent.TryBindOuter(from, reference, out variable)) {
            return true;
        }
    }
    variable = null;
    return false;
}
```

Reply:
I like your idea.

Yes, I tried to keep `BindReference` consistent with other versions, to make it easier to compare them. When I started working on this part of code a few years ago I was often puzzled by (and suspicious of) all those subtle differences.